src/models/dynamic_inverter.jl

```diff
@@ -0,0 +1,84 @@
+abstract type InverterComponent <: DynamicComponent end
+
+mutable struct DynamicInverter{C<:Converter,
+                               O<:OuterControl,
+                               VC<:VSControl,
+                               DC<:DCSource,
+                               P<:FrequencyEstimator,
+                               F<:Filter} <: DynInjection
```

Comment:
Parameterizing on 6 different types seems like a lot of complexity.  Maybe it's required. It also makes me wonder if the problem can be broken down into smaller pieces.
If you plan to use this for multiple dispatch, trying to tell the difference between those function calls with your eyes is going to be difficult.

Reply:
This is because there are too many internal components for an Inverter, a simple inverter already has to make 19 states. But there are too many interactions to separate further.